File: leetcode/ruby/divisible-and-non-divisible-sums-difference.rb

# Leetcode problem 2894: Divisible and Non-divisible Sums Difference.

# @param {Integer} n
# @param {Integer} m
# @return {Integer}
def difference_of_sums(n, m)
  return n * (n + 1) / 2 - m * (n / m) * (n / m + 1)
end

File: leetcode/rust/divisible-and-non-divisible-sums-difference.rs
// Leetcode problem 2894: Divisible and Non-divisible Sums Difference.

struct Solution {}

impl Solution {
    pub fn difference_of_sums(n: i32, m: i32) -> i32 {
        n * (n + 1) / 2 - m * (n / m) * (n / m + 1)
    }
}

fn main() {}
